Controller/ThemeController.php

```diff
@@ -73,7 +88,15 @@ public function switchAction(Request $request)
 
         $this->activeTheme->setName($theme);
 
-        $url = $request->headers->get('Referer', '/');
+
+        if ($this->defaultRoute) {
+          $redirect = $this->router->generate($this->defaultRoute);
```

Comment:
because we cannot use `homepage` as default route, and will probably use hardcoded `/` (see comment to Configuration), you no longer need router in this controller

Reply:
The point of this change is not to use a hardcoded route back to "/", because in some cases Symfony may not run in root domain. In a later commit I added "/" as a fallback in case there is no route set in config, so we should have that covered.

Reply:
Default configuration for route is now "null". When there is a route given in the configuration file, the controller uses the router to redirect back to that route. If no route is given in configuration, it defaults to "null" and the redirect goes to "/" as before. :)